src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,135 @@
+package duke.task;
+
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+
+/**
+ * Represents the list of task in the current session.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Initialise the ArrayList to store the task.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>(100);
+    }
+
+    /**
+     * Generate the tasks in the Arraylist.
+     *
+     * @return tasks List of all task.
+     */
+    public ArrayList<Task> getTasks() {
+        return tasks;
+    }
+
+    /**
+     * Adds a task object directly to the task list.
+     *
+     * @param t Task to add to the list.
+     */
+    public void addTask(Task t) {
+        tasks.add(t);
+    }
+
+    /**
+     * Returns a string representation of the list of tasks.
+     *
+     * @return String representation of the task list
+     */
+    public String listTasks() {
+        String output = "Here are the tasks in your list:\n";
+        for (int i = 0; i < tasks.size(); i++) {
+            output += "     " + (i + 1) + ". " + tasks.get(i).toString();
+            if (i < tasks.size() - 1) {
+                output += "\n";
+            }
+        }
+        return output;
+    }
+
+    /**
+     * Marks a task as done given the task number.
+     *
+     * @param taskNum Task number of task to mark.
+     * @throws DukeException If the task number is invalid.
+     */
+    public void markTask(int taskNum) throws DukeException {
+        if (taskNum < 0 || taskNum >= tasks.size()) {
+            throw new DukeException("Task number invalid");
+        }
+        tasks.get(taskNum).mark();
+    }
+
+    /**
+     * Marks a task as undone given the task number.
+     *
+     * @param taskNum Task number of task to mark.
+     * @throws DukeException If the task number is invalid.
+     */
+    public void unmarkTask(int taskNum) throws DukeException {
+        if (taskNum < 0 || taskNum >= tasks.size()) {
+            throw new DukeException("Task number invalid");
+        }
+        tasks.get(taskNum).unmark();
+    }
+
+    /**
+     * Removes a task given the task number.
+     *
+     * @param taskNum Task number of task to remove.
+     * @return Success message.
+     * @throws DukeException If the task number is invalid.
+     */
+    public Task deleteTask(int taskNum) throws DukeException {
+        if (taskNum < 0 || taskNum >= tasks.size()) {
+            throw new DukeException("Task number invalid");
+        }
+        Task removedTask = tasks.get(taskNum);
+        tasks.remove(taskNum);
+        return removedTask;
```

Comment:
Oh! `remove` for an `ArrayList` acts like `pop`. You don't need to store the Task first!
```suggestion
        return tasks.get(taskNum);
```